pkg/controller/direct/bigqueryconnection/connection_controller.go

```diff
@@ -242,8 +254,9 @@ func (a *Adapter) Update(ctx context.Context, updateOp *directbase.UpdateOperati
 		return err
 	}
 	mapCtx := &direct.MapContext{}
-	desired := a.desired.DeepCopy()
-	connection := BigQueryConnectionConnectionSpec_ToProto(mapCtx, &desired.Spec)
+	connection := BigQueryConnectionConnectionSpec_ToProto(mapCtx, &a.desired.Spec)
+	connection = merge3(connection, a.desired.Status.ObservedState)
```

Comment:
This is a quick fix for GCP error "Error 400: Only 'aws.access_role.iam_role_id' field can be updated." where the `iam_role_id` is a output-only field, that the `CompareProtoMessage` considers the field needs to be changed.

I think the final solution should be a 3 way merge among the desired (`.spec`), previously applied (`.status.observedState`) and the actual.